docs/how-to/hip_runtime_api/memory_management/unified_memory.rst

```diff
@@ -0,0 +1,737 @@
+    }
+
+Memory range attributes
+--------------------------------------------------------------------------------
+
+:cpp:func:`hipMemRangeGetAttribute()` allows you to query attributes of a given
+memory range. The attributes are given in :cpp:enum:`hipMemRangeAttribute`.
+
+.. code-block:: cpp
+    :emphasize-lines: 44-49
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)              \
+    {                                          \
+        const hipError_t err = expression;     \
+        if(err != hipSuccess){                 \
+            std::cerr << "HIP error: "         \
+                << hipGetErrorString(err)      \
+                << " at " << __LINE__ << "\n"; \
+        }                                      \
+    }
+
+    // Addition of two values.
+    __global__ void add(int *a, int *b, int *c) {
+        *c = *a + *b;
+    }
+
+    int main() {
+        int *a, *b, *c;
+        unsigned int attributeValue;
+        constexpr size_t attributeSize = sizeof(attributeValue);
+
+        int deviceId;
+        HIP_CHECK(hipGetDevice(&deviceId));
+
+        // Allocate memory for a, b and c that is accessible to both device and host codes.
+        HIP_CHECK(hipMallocManaged(&a, sizeof(*a)));
+        HIP_CHECK(hipMallocManaged(&b, sizeof(*b)));
+        HIP_CHECK(hipMallocManaged(&c, sizeof(*c)));
+
+        // Setup input values.
+        *a = 1;
+        *b = 2;
+
+        HIP_CHECK(hipMemAdvise(a, sizeof(*a), hipMemAdviseSetReadMostly, deviceId));
+
+        // Launch add() kernel on GPU.
+        hipLaunchKernelGGL(add, dim3(1), dim3(1), 0, 0, a, b, c);
+
+        // Wait for GPU to finish before accessing on host.
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Query an attribute of the memory range.
+        HIP_CHECK(hipMemRangeGetAttribute(&attributeValue,
+                                attributeSize,
+                                hipMemRangeAttributeReadMostly,
+                                a,
+                                sizeof(*a)));
+
+        // Prints the result.
+        std::cout << *a << " + " << *b << " = " << *c << std::endl;
+        std::cout << "The array a is" << (attributeValue == 1 ? "" : " NOT") << " set to hipMemRangeAttributeReadMostly" << std::endl;
+
+        // Cleanup allocated memory.
+        HIP_CHECK(hipFree(a));
+        HIP_CHECK(hipFree(b));
+        HIP_CHECK(hipFree(c));
+
+        return 0;
+    }
+
+Asynchronously attach memory to a stream
+--------------------------------------------------------------------------------
+
+The :cpp:func:`hipStreamAttachMemAsync()` function is able to asynchronously attach
+memory to a stream, which can help concurrent execution when using streams.
+
+Currently, this function is a no-operation (NOP) function on AMD GPUs. It simply
+returns success after the runtime memory validation passed. This function is
+necessary on Microsoft Windows, and HMM is not supported on this operating
```

Comment:
```suggestion
required on Microsoft Windows, but HMM is not supported on this operating
```

It indicates that it is necessary on Microsoft Windows, but not needed on Linux? Therefore it doesn't do anything on Linux or isn't available on Linux? I understand HMM is not available on Windows, so using it on Windows results in nothing, but HMM is supported on Linux, so using it on Linux does nothing?

Reply:
Good point, I didn't refactor this paragraph properly. I rewrote the description of what this function does to give a better understanding, and removed the paragraph about the support completely - HIP does not support unified memory on windows at all currently, and this function has not yet been implemented in Linux, so it is just for passing it along to CUDA based systems. That however is explained in the reference to this function, so it does not need to be repeated here.